runtime/interpreter/interpreter.go

```diff
@@ -242,14 +242,15 @@ type Storage interface {
 type ReferencedResourceKindedValues map[atree.StorageID]map[ReferenceTrackedResourceKindedValue]struct{}
 
 type Interpreter struct {
-	Location     common.Location
-	statement    ast.Statement
-	Program      *Program
-	SharedState  *SharedState
-	Globals      GlobalVariables
-	activations  *VariableActivations
-	Transactions []*HostFunctionValue
-	interpreted  bool
+	Location            common.Location
+	statement           ast.Statement
+	Program             *Program
+	SharedState         *SharedState
+	Globals             GlobalVariables
+	activations         *VariableActivations
+	Transactions        []*HostFunctionValue
+	interpreted         bool
+	cachedIntegerValues map[StaticType]map[int64]IntegerValue
```

Comment:
This was in response to https://github.com/onflow/cadence/pull/2523#discussion_r1246912655. I should have maybe been clearer about the particular implementation of the lookup table.

The table should not be per interpreter, because there are many interpreters (one per location), but can be global. We can start using it here, but later also use if for other cases ("small integer optimization")

Reply:
If it's global, I think it would be better to have a limit on the range of integers that are memoized (e.g. precompute/memoize only if `n<10`, etc.). Otherwise (currently) it can store an infinite number of values which is a bit dangerous.

Reply:
Yes, it definitely should have a limit and it should be small. Currently, only 0 and 1 are used AFAICS.
I think Java uses range -128 to 127. So maybe just change the key type to `int8`?

Reply:
Done, made it global and key type to int8.